src/main/java/fact/io/FITSWriter.java

```diff
@@ -69,9 +73,29 @@
     @Override
     public Data process(Data item) {
         Data outputItem = DataFactory.create();
+        if (!allowNullKeys) {
```

Comment:
Maybe make this a method of a writer interface? So we avoid code duplication and can use it in the FITSWriter and the JSONWriter?